File: my_python_pkg/my_python_pkg/smartphone.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SmartphoneSubscriber(Node):

    def __init__(self):
        super().__init__('smartphone_subscriber')
        self.subscription = self.create_subscription(String,'robot_news', self.callback_robot_news, 10)
        self.get_logger().info('Smartphone Subscriber has been started')

    def callback_robot_news(self, msg):
        self.get_logger().info(msg.data)

def main(args=None):
    rclpy.init(args=args)
    smartphone_subscriber = SmartphoneSubscriber()
    rclpy.spin(smartphone_subscriber)
    smartphone_subscriber.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class SmartphoneNode : public rclcpp::Node
{
public:
    SmartphoneNode() : Node("smartphone")
    {
        subscriber_ = this->create_subscription<std_msgs::msg::String>(
            "robot_news", 10, std::bind(&SmartphoneNode::callbackRobotNews, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "Smartphone has been started.");
    }

private:
    void callbackRobotNews(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Receiving '%s'", msg->data.c_str());
    }

    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscriber_; 
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);                         // Initialize the ROS 2 system
    auto node = std::make_shared<SmartphoneNode>();   // Create an instance of SmartphoneNode
    rclcpp::spin(node);                               // Keep the node running
    rclcpp::shutdown();                               // Shutdown ROS 2
    return 0;
}
